### PR TITLE
Enable whitespace-related Checkstyle checks

### PR DESCRIPTION
**Pull request by basil**
Much of the early code written by Kohsuke was very lean on whitespace. This neither reflects modern Java coding conventions nor most of the other code written in this project since Kohsuke's time. This PR enables some whitespace-related Checkstyle rules and fixes existing violations. I've intentionally tried to stay away from anything controversial and simply adopted some basic checks to reflect the consensus throughout most of the repository. For the most part, the only code that I've changed has been old Kohsuke code. I don't expect that people following modern Java coding style should trip on these checks.

I suggest reviewing this change with "Hide whitespace" enabled in the GitHub UI.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: pom.xml
Comment (basil) [inline pom.xml]: The main non-whitespace portion of this PR.

**Review by timja — APPROVED** [review]
I checked the first 30 or so files all looks sensible

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
